### PR TITLE
extmod/utimeq: Fix possible redefinition of DEBUG macro

### PR DESCRIPTION
**Pull request by stinos**
DEBUG is already 'reserved' as a global preprocessor definition because
it is used in nlr.h. For instance the msvc port defines DEBUG on the
compiler commandline in debug builds.
As such defining it again might emit compiler warnings and is confusing.
So, in modutimeq, rename DEBUG to MICROPY_UTIMEQ_DEBUG.

@pfalcon I'm not sure if there is a convention for names of per-file local macros so I just picked one with similar naming as the global ones.

### CODE REVIEW

**Comment by pfalcon** [comment]
I'm not sure why it's used in nlr.h. In modutimeq, "DEBUG" is used in the same way as "DEBUG_printf" in bunch of other files, as a local macro. So, we need to find consistent solution for this.

**Comment by dpgeorge** [comment]
> I'm not sure why it's used in nlr.h.

Yeah, it's a bit strange and I don't know why nlr.h uses DEBUG.  If it wants to be a global then probably it should be NDEBUG because that is what assert uses.

So I propose changing it to NDEBUG (this will impact unix build because assertions are enabled by default) or MICROPY_DEBUG_NLR.  Since this debugging feature is not really needed/used anymore it makes sense to have it MICROPY_DEBUG_NLR so it's off by default.

**Comment by stinos** [comment]
Ok changed to MICROPY_DEBUG_NLR and fixed an unused instance of DEBUG in objint_longlong.c. And removed it from the msvc build.

File: py/objint_longlong.c [hunk 1]
```diff
@@ -40,14 +40,6 @@
 
 #if MICROPY_LONGINT_IMPL == MICROPY_LONGINT_IMPL_LONGLONG
 
-// Python3 no longer has "l" suffix for long ints. We allow to use it
```
Comment (pfalcon) [inline py/objint_longlong.c]: Why is this removed?

Comment (stinos) [inline py/objint_longlong.c]: See commit comment: hasn't been used in years

Comment (pfalcon) [inline py/objint_longlong.c]: Well, commit message has a surface bug ("stale" duplicated) and ungrounded assumption of "hasn't been used in years". We just recently found yet another bug in long int handling, and this diagnostic measure is there exactly to allow to debug such cases easier. So I'm not sure if it "hasn't been used in years", but everyone certainly should be using it every time needed and it shouldn't be removed, certainly not as part of unrelated changes.

Comment (stinos) [inline py/objint_longlong.c]: I just saw a definition which was last used in a function which got removed and it wasn't clear to me from the comment that the point of SUFFIX is to arbitrarily use it when needed for debugging.
Anyway: proper fix is then to also introduce e.g. MICROPY_DEBUG_LONGINT ?

Comment (pfalcon) [inline py/objint_longlong.c]: Ah, so I finally got the "unused anywhere" part, sorry. Still, this shows why unrelated changes should be included in one commit - this spurs confusion.

> Anyway: proper fix is then to also introduce e.g. MICROPY_DEBUG_LONGINT ?

Well, so I looked into this just to not let it linger for months - we do have technical debt, so it can only be appreciated when non-maintainers try to decrease it. But: "proper" way to fix needs to be considered for months, because it's wide issue, and not a priority at all. I can tell you an anecdote that I was unpleasantly surprised to see in Contiki the situation when you need to go and patch a particular file to enable debugging there. Fast forward few years, and we're using the same "anti-pattern" in uPy. That's probably because this "anti-pattern" is awfully practical, and any alternative while solves something, brings in new issues, so even the final balance isn't clear, and that's what needs consideration for months.

Anyway, to be done with it, please fix typo in commit message, and let's have this merged.

Comment (stinos) [inline py/objint_longlong.c]: > this shows why unrelated changes should be included in one commit

Yes sorry, at the time of writing it looked all very related and non-confusing to me as I just spent some time familiarizing with all aspects of it. But in hindsight it is not all that obvious.

Commit message is fixed and commits rebased.

> That's probably because this "anti-pattern" is awfully practical

Indeed, in absence of fully configurable logging with named loggers and filtering etc, it is imo not really an anti-pattern anymore and it works very well. The opposite can also be an anti-pattern: I used some library once (don't remember which one) where enabling one single definition globally resulted in spitting out thousands of log lines. Finding the ones of interest isn't exactly a pleasure then.

Comment (pfalcon) [inline py/objint_longlong.c]: Right, and so the bottom line is that MICROPY_DEBUG_NLR is a one-off case, not a new pattern we follow (until it will undergo enough discussion and dubbed to be so, which discussion is not a priority).

**Comment by pfalcon** [comment]
Merged, thanks!
